emcc.py

```diff
@@ -2683,12 +2688,14 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
     if DEBUG:
       save_intermediate_with_wasm('pre-eval-ctors', wasm_binary_target)
     shared.Building.eval_ctors(final, wasm_binary_target, binaryen_bin, debug_info=debug_info)
+
   # after generating the wasm, do some final operations
-  if shared.Settings.SIDE_MODULE:
+  if shared.Settings.SIDE_MODULE and not shared.Settings.WASM_BACKEND:
```

Comment:
how will this work with the wasm backend? will it emit a library directly from lld?

Reply:
Yes, this change might make more sense the follow up CL I guess.   I the plan is to have lld produce a dylink with `-shared`